aloha-vw-jni/src/main/scala/com/eharmony/aloha/models/vw/jni/vwJniModelSources.scala

```diff
@@ -0,0 +1,116 @@
+package com.eharmony.aloha.models.vw.jni
+
+import java.io.{File, FileOutputStream}
+
+import com.eharmony.aloha.io.fs.FsInstance
+import org.apache.commons.codec.binary.Base64
+import org.apache.commons.io.IOUtils
+import vw.VW
+
+
+object VwJniModelSource {
+    private[jni] val InitialRegressorPresent = """^(.*\s)?-i.*$"""
+}
+
+/**
+ * Needs to be Serializable
+ */
+sealed trait VwJniModelSource extends Serializable {
+    import VwJniModelSource._
+    def params: String
+    def timeMillis: Long
+    def copyContentToLocalIfNecessary(dest: File): Unit
+    def shouldDeleteLocalFile: Boolean
+
+    def localFile(modelId: Long): File
+
+    final def vwModel(modelId: Long): VW = {
+        val locFile = localFile(modelId)
+        val shouldDelete = shouldDeleteLocalFile
+
+        if (shouldDelete)
+            locFile.deleteOnExit()
+
+        copyContentToLocalIfNecessary(locFile)
+
+        val newParams = updatedVwModelParams(modelId)
+        val vw = new VW(newParams)
+
+        if (shouldDelete)
+            locFile.delete()
+
+        vw
+    }
+
+    def updatedVwModelParams(modelId: Long): String = {
+        val initialRegressorParam = s" -i ${localFile(modelId).getCanonicalPath} "
+
+        if (params matches InitialRegressorPresent)
+            throw new IllegalArgumentException(s"For model $modelId, initial regressor (-i) vw parameter supplied and model provided.")
+
+        initialRegressorParam + params
+    }
+
+    def tmpDir() = {
+        val t = File.createTempFile("abc123", "zyx987")
```

Comment:
Because the stupid func requires minimum length strings but does really explain to much so i just chose some thing work because that file is deleted immediately and is just used ro determine the temp directory



On Aug 21, 2015, 4:36 PM, at 4:36 PM, jmorra <notifications@github.com> wrote:
>> +            locFile.delete()
>> +
>> +        vw
>> +    }
>> +
>> +    def updatedVwModelParams(modelId: Long): String = {
>> +        val initialRegressorParam = s" -i
>${localFile(modelId).getCanonicalPath} "
>> +
>> +        if (params matches InitialRegressorPresent)
>> +            throw new IllegalArgumentException(s"For model $modelId,
>initial regressor (-i) vw parameter supplied and model provided.")
>> +
>> +        initialRegressorParam + params
>> +    }
>> +
>> +    def tmpDir() = {
>> +        val t = File.createTempFile("abc123", "zyx987")
>
>Why these Strings, can we use something that has at least some meaning?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/eHarmony/aloha/pull/35/files#r37687506